src/library/searchquery.cpp

```diff
@@ -471,6 +471,221 @@ double DurationFilterNode::parse(const QString& arg, bool* ok) {
     return 60 * m + s;
 }
 
+// static
+constexpr double BpmFilterNode::kRelativeRangeDefault;
+
+// static
+double BpmFilterNode::s_relativeRange = kRelativeRangeDefault;
+
+// static
+void BpmFilterNode::setBpmRelativeRange(double range) {
+    // range < 0 would yield zero results because m_dRangeLow > m_dRangeHigh
+    VERIFY_OR_DEBUG_ASSERT(range >= 0) {
+        return;
+    }
+    s_relativeRange = range;
+}
+
+BpmFilterNode::BpmFilterNode(QString& argument, bool fuzzy, bool negate)
+        : m_fuzzy(fuzzy),
+          m_negate(negate),
+          m_isNullQuery(false),
+          m_isOperatorQuery(false),
+          m_isRangeQuery(false),
+          m_isHalfDoubleQuery(false),
+          m_operator("="),
+          m_bpm(0.0),
+          m_rangeLower(0.0),
+          m_rangeUpper(0.0),
+          m_bpmHalfLower(0.0),
+          m_bpmHalfUpper(0.0),
+          m_bpmDoubleLower(0.0),
+          m_bpmDoubleUpper(0.0) {
+    if (argument == kMissingFieldSearchTerm) {
+        m_isNullQuery = true;
+        return;
+    }
+    qWarning() << "     .BpmFilter:" << argument;
```

Comment:
As written, this will create a lot of warning spam.  In other modules we have a localized debug option for debugging that specific feature by namespace (see EngineBuffer).

Reply:
yes, the last commit is for developing and I'll remove that commit
(usually I mention this in the PR TODO, recently I've been abusing "fixup!" to block accidental merge)